main.go

```diff
@@ -1,65 +1,83 @@
 package main
 
 import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
 	"net/http"
-	"os"
-	"time"
+	"errors"
+	"github.com/gin-gonic/gin"
+	"io/ioutil"
+	"encoding/json"
 )
 
 type WeatherData struct {
 	City string `json:"city"`
-	Info string `json:"info"`
+	Country string `json:"country"`
+	Temp_C float64 `json:"temp_c"`
+	Temp_F float64 `json:"temp_f"`
+	Wind_KPH float64 `json:"wind_kph"`
+	Wind_MPH float64 `json:"wind_mph"`
 }
 
-var apiKey = os.Getenv("bc94089fef9d46968ff8d724889f9fff")
+func getWeatherByCity(city string) (*WeatherData, error) {
+	url := "https://weatherapi-com.p.rapidapi.com/current.json?q=" + city
 
-func main() {
-	http.HandleFunc("/weather/", func(w http.ResponseWriter, r *http.Request) {
-		city := r.URL.Path[len("/weather/"):]
+	req, _ := http.NewRequest("GET", url, nil)

```

Comment:
Consider caching the result for a while. This will still give good enough results (weather doesn't change super fast!), and you can avoid being rate limited by the weather service